src/data-type.js

```diff
@@ -295,19 +295,20 @@ const TYPE = module.exports.TYPE = {
     writeParameterData: function(buffer, parameter, options) {
       if (parameter.value != null) {
         let days, dstDiff, milliseconds, seconds, threeHundredthsOfSecond;
+        var time = new Date(+parameter.value);
```

Comment:
What is the rationale behind this conversion?
If `parameter.value` is already a `Date`, the `+` operator converts it to an integer and the `new Date()` converts it back to a `Date`. 
If `parameter.value` is a string that does not contain a number, the `+` operator produces a NaN.
e.g.: `+'2017-10-11'` ==> `NaN`
or: `+'Wed Oct 11 2017 19:49:46 GMT+0200'` ==> `NaN`